packages/react-components/src/graph-visualizer/graph.ts

```diff
@@ -0,0 +1,523 @@
+/* eslint-disable prefer-const */
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+import cytoscape, { EventObjectCore, EventObjectEdge } from 'cytoscape';
+import type {
+  Css,
+  EdgeDefinition,
+  ElementsDefinition,
+  LayoutOptions,
+  EventObjectNode,
+  NodeCollection,
+  NodeDefinition,
+  NodeSingular,
+} from 'cytoscape';
+// `cytoscape-cise` does not have TypeScript definitions.
+//@ts-ignore
+import cise from 'cytoscape-cise';
+import type { SetRequired, ValueOf } from 'type-fest';
+
+import type { Health } from './lib/types';
+import { isNil, isString } from './lib/utils/lang';
+import { GRAPH_COLORS } from './constants';
+import { createSvgString, getPolygonPoints, getSvgSize } from './svg';
+import type {
+  EdgeArrowStyleProps,
+  EdgeData,
+  EdgeEndpoint,
+  EdgeStyleProps,
+  EventDetail,
+  NodeData,
+  NodeRenderData,
+  NodeStyleProps,
+  Subscriber,
+} from './types';
+//import { verify } from "crypto";
+
+interface NodesDefinition {
+  nodes: NodeDefinition[];
+}
+
+const EDGE_DEFAULT_STYLE_PROPS: SetRequired<EdgeStyleProps, 'targetArrow'> = {
+  color: GRAPH_COLORS.GRAY_40,
+  targetArrow: {
+    color: GRAPH_COLORS.GRAY_40,
+    scale: 1.5,
+    shape: 'triangle',
+  },
+  width: 3,
+};
+
+const EDGE_HOVER_STYLE_PROPS: EdgeStyleProps = {
+  ...EDGE_DEFAULT_STYLE_PROPS,
+  color: GRAPH_COLORS.GRAY_10,
+  targetArrow: {
+    ...EDGE_DEFAULT_STYLE_PROPS.targetArrow,
+    color: GRAPH_COLORS.GRAY_10,
+  },
+  width: 3,
+};
+
+const EDGE_SELECTED_STYLE_PROPS: EdgeStyleProps = {
+  ...EDGE_HOVER_STYLE_PROPS,
+  width: 6,
+};
+
+const NODE_DEFAULT_STYLE_PROPS: NodeStyleProps = {
+  backgroundImage: getNormalSvg,
+  borderWidth: 3,
+  labelTextColor: GRAPH_COLORS.GRAY_14,
+  labelTextSize: 16,
+  labelOffset: 12,
+  labelPadding: 8,
+  labelPosition: 'bottom',
+  shape: getNodeShape,
+  transitionDuration: 300,
+};
+
+const NODE_HOVER_STYLE_PROPS: NodeStyleProps = {
+  ...NODE_DEFAULT_STYLE_PROPS,
+  backgroundImage: getHoverSvg,
+  borderColor: GRAPH_COLORS.WHITE,
+  labelTextColor: GRAPH_COLORS.WHITE,
+};
+
+const NODE_SELECTED_STYLE_PROPS: NodeStyleProps = {
+  ...NODE_DEFAULT_STYLE_PROPS,
+  backgroundImage: getSelectedSvg,
+  borderColor: GRAPH_COLORS.WHITE,
+  borderWidth: 6,
+  labelTextColor: GRAPH_COLORS.WHITE,
+};
+
```

Comment:
I think all these prop functions can be split to a separate file.

Reply:
Addressed, moved to a separate defaultProps file.